### DISCUSSION TITLE
how should be specified and represent a nested array of a query parameter ?

### DISCUSSION THREAD

**Discussion by RSH17** [OAI/OpenAPI-Specification#3052] [Q&A] [answered]
- As long as I understand, the query parameter type can define as a nested array.

 `{
            "name": "petName",
            "in": "query",
            "description": "Status of pet that needs to be updated",
            "schema": {
              "type": "array",
              "items": {
                "type": "array",
                  "items": {
                    "type": "array",
                      "items": {
                        "type": "integer"
                      }
                  }
              }
            }
     }`

![Capture1](https://user-images.githubusercontent.com/56576410/195317322-bd742c0b-6311-4941-a8e6-327a55edef48.PNG)

- But this will represent a flat array in the request URL of the swaggers. 

`https://petstore.swagger.io/v2/pet/19?status=available&petName=5&petName=5&petName=66&petName=77`

![Capture3](https://user-images.githubusercontent.com/56576410/195317007-09ca1cf6-4acc-4370-adb3-e0eba0df0c60.PNG)
![Capture4](https://user-images.githubusercontent.com/56576410/195317278-f05042e7-4f6d-4560-999a-6505969068e7.PNG)

I was wondering if this scenario is valid or if could you please give me a brief introduction about how to specify and represent a nested array of a query parameter.

Best regards

**Comment by hkosova** [comment, marked as answer]
As of OpenAPI 3.1, nested arrays and nested objects are **not** actually supported in query parameters because the serialization rules are not defined for nested structures. More info here:

* #1706
* [OpenAPI path/query parameters nested structure serialization](https://stackoverflow.com/questions/67745944/openapi-path-query-parameters-nested-structure-serialization) on Stack Overflow

**Reply by RSH17** [reply]
Thanks.

**Comment by handrews** [comment]
Looks like this was resolved, closing.
